Gems/Terrain/Code/Source/TerrainRenderer/TerrainMeshManager.cpp

```diff
@@ -84,6 +95,28 @@ namespace Terrain
         AZ::RPI::ShaderSystemInterface::Get()->SetGlobalShaderOption(AZ::Name{ "o_useTerrainClod" }, AZ::RPI::ShaderOptionValue{ m_config.m_clodEnabled });
     }
 
+    bool TerrainMeshManager::UpdateGridSize(float distanceToFirstLod)
+    {
+        float queryResolution = 1.0f;
+        AzFramework::Terrain::TerrainDataRequestBus::BroadcastResult(
+            queryResolution, &AzFramework::Terrain::TerrainDataRequests::GetTerrainHeightQueryResolution);
+
+        float quadsToFirstLod = distanceToFirstLod / queryResolution;
+        uint32_t quadsPerSector = aznumeric_cast<uint32_t>(quadsToFirstLod / 4.0f);
+        uint32_t gridSize = AZ::RHI::IsPowerOfTwo(quadsPerSector) ? quadsPerSector : (AZ::RHI::NextPowerOfTwo(quadsPerSector) >> 1);
+        gridSize = AZStd::GetMin(gridSize, 128u); // x/y positions must be able to fix in 8 bits (256 is too large by 1)
+        gridSize = AZStd::GetMax(gridSize, 8u); // make sure there's enough vertices to be worth drawing.
```

Comment:
Suggestion: `gridSize = AZ::GetClamp(gridSize, 8u, 128u);`

Reply:
I kinda lean towards keeping them separate since the reasons for the min and max a different as the comments explain. Should be no different computationally.

Reply:
That's totally reasonable. :)